.ci/Jenkinsfile

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+
+def NODE_JS_VERSIONS = [8,10,12]
+def nodeJsVersion = NODE_JS_VERSIONS[randomNumber(min: 0, max:2)]
+
+pipeline {
+  agent {
+    label 'docker && immutable'
+  }
+
+  environment {
+    REPO = 'elasticsearch-js'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NODE_JS_DEFAULT_VERSION = "${nodeJsVersion}"
+    NODE_JS_VERSIONS = "${NODE_JS_VERSIONS.join(',')}"
+    HOME = "${env.WORKSPACE}"
+    npm_config_cache = 'npm-cache'
+  }
+
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+
+  triggers {
+    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+    // changeRequest() will return true in case of a commit or a pr
+    // we will have a daily cron job only ofr branches that don't have an active pr
+    cron(changeRequest() ? '' : '@daily')
+  }
+
+  stages {
+    stage('Checkout') {
+      options { skipDefaultCheckout() }
+      steps {
+        withGithubNotify(context: 'Checkout') {
+          deleteDir()
+          gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: false)
+          stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+        }
+      }
+    }
+
+    stage('Install dependencies') {
+      options { skipDefaultCheckout() }
+      steps {
+        withGithubNotify(context: 'Install dependencies') {
+          deleteDir()
+          unstash 'source'
+          script {
+            buildDockerImage(image: "node:${env.NODE_JS_DEFAULT_VERSION}-alpine").inside(){
+              dir("${BASE_DIR}"){
+                sh(label: 'System info', script: '''node --version
```

Comment:
Cosmetic change: you can use a single line with `;` between both commands if you don't like the multine